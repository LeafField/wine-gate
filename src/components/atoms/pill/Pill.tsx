import React, { FC } from "react";

type Props = {
  title: string;
  big?: boolean;
};

const Pill: FC<Props> = ({ title, big = false }) => {
  return (
    <div
      className={`w-fit rounded-[3.2rem] border border-black px-2 py-[0.125rem] text-black ${big ? "text-base" : "text-sm"}`}
    >
      {title}
    </div>
  );
};

export default Pill;
