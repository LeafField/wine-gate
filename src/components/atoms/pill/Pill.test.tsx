import React from "react";
import { render } from "@testing-library/react";
import Pill from "./Pill";
import { describe, test, expect } from "vitest";

describe("Pillの単体テスト", () => {
  test("渡されたPropsに対して正しくレンダリングされているか", () => {
    const { getByText } = render(<Pill title="渋い" />);
    expect(getByText("渋い")).toBeDefined();
  });

  test("スナップショット", () => {
    const { container } = render(<Pill title="渋い" />);
    expect(container).toMatchSnapshot();
  });
});
