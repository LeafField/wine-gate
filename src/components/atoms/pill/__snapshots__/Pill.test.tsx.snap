// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pillの単体テスト > スナップショット 1`] = `
<div>
  <div
    class="w-fit rounded-[3.2rem] border border-black px-2 py-[0.125rem] text-sm text-black "
  >
    渋い
  </div>
</div>
`;
